server/opendp_apps/analysis/tools/dp_histogram_spec.py

```diff
@@ -0,0 +1,191 @@
+                      ci=CI_95.
+                      fixed_value=1)
+    """
+    def __init__(self, props: dict):
+        """Set the internals using the props dict"""
+        # TODO: Generates an error when this isn't initialized somewhere
+        super().__init__(props)
+
+    def additional_required_props(self):
+        """
+        Add a list of required properties
+        example: ['min', 'max']
+        """
+        return ['min', 'max']
+
+    def run_01_initial_handling(self):
+        """
+        """
+        if self.fixed_value is not None:
+            # attempt conversion to int
+            if not self.convert_to_int('fixed_value'):
+                return
+
+        # TODO: These default values are allowing the tests to pass,
+        #  but we need to process cases where min and max are referring to counts of a
+        #  categorical variable.
+        if not self.min:
+            self.min =  0.0
+        elif self.convert_to_int('min') is False:
+            return
+
+        if not self.max:
+            self.max = 10.0
+        elif self.convert_to_int('max') is False:
+            return
+
+
+    def run_03_custom_validation(self):
+        """
+        This is a place for initial checking/transformations
+        such as making sure values are floats
+        Example:
+        self.check_numeric_fixed_value()
+        """
+        if self.has_error():
+            return
+
+        self.check_numeric_fixed_value()
+
+    def check_scale(self, scale, preprocessor, dataset_distance, epsilon):
+        """
+        Return T/F
+        :param scale:
+        :param preprocessor:
+        :param dataset_distance:
+        # :param epsilon:
+        :return:
+        """
+        if self.has_error():
+            return
+
+        return (preprocessor >> make_base_geometric(scale)).check(dataset_distance, epsilon)
+
+    def get_preprocessor(self):
+        """
+        Set up and return computation chain
+        :return:
+        """
+        if self.has_error():
+            return
+
+        # Have we already already assembled it?
+        #
+        if self.preprocessor is not None:
+            # Yes!
+            return self.preprocessor
+
+        try:
+            preprocessor = (
+                # Convert data into Vec<Vec<String>>
+                    # Selects a column of df, Vec<str>
+                    make_select_column(key=self.col_index, TOA=str) >>
+                    # Cast the column as Vec<Int>
+                    # make_cast(TIA=str, TOA=str) >>
+                    # Impute missing values to 0
+                    # make_impute_constant(self.fixed_value) >>
+                    make_count_by_categories(categories=self.categories, MO=L1Distance[float])
+                # make_base_geometric(scale=1., bounds=(0,201), D="VectorDomain<AllDomain<i32>>")
+            )
+
+            # self.scale = binary_search(lambda s: self.check_scale(s, preprocessor, 1, self.epsilon), bounds=(0, 1000))
```

Comment:
The `self.epsilon` value doesn't seemed to be used; e.g. and `check_accuracy` is setting accuracy.
(see dp_mean_spec.py, e.g. the `laplacian_scale_to_accuracy` and other methods may not have been available eaerlier)

Reply:
Thanks for the reminder on this - I had commented out the noise mechanism a while back. I've just updated to opendp 0.3.0 and added the latest changes in.